src/cluster.py

```diff
@@ -45,7 +45,14 @@ def add_node_to_cluster(self, event: ops.framework.EventBase) -> None:
         """Add node to microceph cluster."""
         if not event.unit:
             return
-        cmd = ["microceph", "cluster", "add", event.unit.name]
+        # get hostname using unit name.
+        hostnames = self.charm.peers.get_all_unit_values(
+            key=event.unit.name, include_local_unit=True
+        )
```

Comment:
One doubt: should we include the local unit here?

Reply:
We are saving the hostnames of each unit against the unit names. Since we use the unit name from event, we get only the hostname we care for and hence it is OK to include local_unit and always use the first element of the list.